int32 state
int32 secsRemaining
